### ISSUE TITLE
QgsJsonExporter doesn't preserve attributes order in QGIS

### ISSUE THREAD

**Issue by trsbar** [qgis/QGIS#41290]
The preservation of the order of attributes is not preserved when exporting a layer in GeoJSON

```
layer = iface.mapCanvas().currentLayer()
exp = QgsJsonExporter(layer)
geojson = exp.exportFeatures(layer.getFeatures())
```

**Comment by rduivenvoorde** [comment]
@trsbar I do not think this is a bug? 

See for example: https://macwright.com/2015/03/23/geojson-second-bite.html (and search for "The order of properties in GeoJSON isn’t defined"). 

Or do you have some quote from the geojson specs that they should? 
https://tools.ietf.org/html/rfc7946 (geojson)
https://tools.ietf.org/html/rfc7464 (json)

**Comment by trsbar** [comment]
That's true, but why are properties always rearranged alphanumerically? Maybe it's not an issue (but a feature request ?)... when you display your data, there is an intereest to keep the order.

**Comment by nboisteault** [comment]
I observed the same behavior with QGIS Server 3.22.3 and 3.16.16 when requesting GeoJSON via WFS. Order is preserved with GML.
I have a small project illustrating this behavior, should I open an issue?
